op-node/p2p/sync.go

```diff
@@ -48,16 +48,16 @@ const (
 	maxThrottleDelay = time.Second * 20
 	// Do not serve more than 20 requests per second
 	globalServerBlocksRateLimit rate.Limit = 20
-	// Allow up to 5 concurrent requests to be served, eating into our rate-limit
-	globalServerBlocksBurst = 5
-	// Do not serve more than 5 requests per second to the same peer, so we can serve other peers at the same time
-	peerServerBlocksRateLimit rate.Limit = 5
-	// Allow a peer to burst 3 requests, so it does not have to wait
-	peerServerBlocksBurst = 3
+	// Allows a burst of 2x our rate limit
+	globalServerBlocksBurst = 40
+	// Do not serve more than 4 requests per second to the same peer, so we can serve other peers at the same time
+	peerServerBlocksRateLimit rate.Limit = 4
+	// Allow a peer to request 30s of blocks at once
+	peerServerBlocksBurst = 15
 	// If the client hits a request error, it counts as a lot of rate-limit tokens for syncing from that peer:
 	// we rather sync from other servers. We'll try again later,
 	// and eventually kick the peer based on degraded scoring if it's really not serving us well.
-	clientErrRateCost = 100
+	clientErrRateCost = peerServerBlocksBurst
```

Comment:
hmm, this works if we assume all nodes use these same values, but if we want to tolerate variations then we probably want to wait a decent chunk of time to ensure that we leave enough time for the peer to resume serving us requests.  Given `WaitN` doesn't allow waiting for capacity beyond the burst limit, we probably need to do the wait in a loop or maybe wait until the full burst amount is available again plus an exponential backoff if we get repeated rate limit errors.

Reply:
So this is in our internal rate limit for requesting for peers, but yes, we may want to use a different mechanism to back off in the case that there is an error.

Reply:
From the comment, my understanding is that this value is used to get us to stop making new requests from a peer for a while after we get an error from the client that we've exceeded their rate limiting. So with this reduced to just `peerServerBlocksBurst` we'll only wait until our counter for that peer reaches zero then try again.  If the remote peer has a lower limit or different time window to us though, we'll get another rejection for exceeding their request limit. So we want to wait a longer period to ensure the remote side will allow us to start making requests again - hence this was originally set much higher than the burst limit.
It looks like we're using this value to back off after any error, not just rate exceeded errors, but I suspect we still want to be waiting a reasonable period of time whereas I think we're now making that back-off period quite short.

Reply:
Yea. We'll consume the full burst & pretty soon be able to retry (250ms until the bucket has a single token in it). We should fix it here & I'll create a ticket to track the full fix.